### PR TITLE
Feature: redux

### PR DESCRIPTION
**Pull request by velvetkevorkian**
Implements a shared store for federated components.
- Add redux state to app-shell and header
- add middleware for passing initial state through prerender
- federate MFE reducers and construct initial state and store in app-shell

### CODE REVIEW

File: apps/AppShell/src/utils/combineInitialStates/index.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const combineInitialState = (initalState: any, state: any) => {
+  const allstates: any = {}
+
+  const keys = Object.keys(initalState)
+  allstates[keys[0]] = initalState[keys[0]]
+
+  Array.from(state).forEach((s: any) => {
+    if (s.dataset.state) {
+      const obj = JSON.parse(s.innerText.replaceAll('&quot;', '"'))
```
[the remark this answers is not in the document]
Comment (velvetkevorkian) [reply apps/AppShell/src/utils/combineInitialStates/index.ts]: Fixed

File: libs/framework/middlewares/src/prerender.tsx [hunk 2]
```diff
@@ -19,6 +21,12 @@ export const prerenderMiddleware = remoteEntry => {
         get: () => () => React,
       },
     },
+    // TODO: can this be automated?
+    'react-redux': {
+      '7.2.8': {
```
[the remark this answers is not in the document]
Comment (velvetkevorkian) [reply libs/framework/middlewares/src/prerender.tsx:26]: #102

File: libs/framework/remote-urls/src/index.tsx [hunk 3]
```diff
@@ -1,6 +1,13 @@
...
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const remotes = JSON.parse(process.env.REMOTE_URLS!)
+    Object.keys(remotes).forEach(remote => {
+      if (remotes[remote].includes('@')) {
+        remotes[remote] = remotes[remote].split('@')[1]
+      }
+    })
+    return remotes
   } catch {
     return {}
```
[the remark this answers is not in the document]
Comment (velvetkevorkian) [reply libs/framework/remote-urls/src/index.tsx:15]: fixed

File: apps/MFE/Header/public/app.html [hunk 4]
```diff
@@ -11,7 +11,6 @@
       window.ssrClientSettings = window.ssrClientSettings || {}
       window.ssrClientSettings.appshell = {
```
[the remark this answers is not in the document]
Comment (velvetkevorkian) [reply apps/MFE/Header/public/app.html]: fixed

**Review by AlzAmido — APPROVED** [review]

**Pull request merged**